x/slashing/keeper/signing_info.go

```diff
@@ -72,6 +72,58 @@ func (k Keeper) SetValidatorMissedBlocks(ctx sdk.Context, address sdk.ConsAddres
 	store.Set(types.ValidatorMissedBlockBitArrayKey(address), bz)
 }
 
+// Get a boolean representing whether a validator missed a block with a specific index offset
+func (k Keeper) GetBooleanFromBitGroups(bitGroupArray []uint64, index int64) bool {
+	// convert the index into indexKey + indexShift
+	indexKey := index / 64
+	indexShift := index % 64
+	if indexKey >= int64(len(bitGroupArray)) {
+		return false
+	}
+	// shift 1 by the indexShift value to generate bit mask (to index into the bitGroup)
+	indexMask := uint64(1) << indexShift
+	// apply the mask and if the value at that `indexShift` is 1 (indicating miss), then the value would be non-zero
+	missed := (bitGroupArray[indexKey] & indexMask) != 0
+	return missed
+}
+
+// Set the missed value for whether a validator missed a block
+func (k Keeper) SetGetBooleanInBitGroups(bitGroupArray []uint64, index int64, missed bool) []uint64 {
```

Comment:
nit - can we rename this to just `SetBooleanInBitGroups`, or am i missing something with the "Get" part?

Reply:
nah this is a typo